krabs/krabs/trace.hpp

```diff
@@ -348,7 +354,8 @@ namespace krabs {
     template <typename T>
     trace<T>::~trace()
     {
-        stop();
+        // Never throw in destructor
+        try { stop(); } catch (...) {}
```

Comment:
Did you see this throw? I'd need to look to see how trace_manager can throw.
Usually, we don't want to catch everything. There's also implications for SEH callbacks that might have been set up.

Reply:
Throwing an exception out of a destructor is dangerous. If another exception is already propagating the application will terminate.
ControlTrace(EVENT_TRACE_CONTROL_STOP) or CloseTrace() can return error inside stop(). Using SEH is to much.

Reply:
I'm aware that destructors shouldn't throw, however it's not a good pattern to just add "catch all" blocks in all the destructors. Catch what we expect.

Reply:
Ok. If we add error_callback to trace object it can solve a problem with user notification.